base/uk.ac.stfc.isis.ibex.journal/src/uk/ac/stfc/isis/ibex/journal/JournalSqlStatement.java

```diff
@@ -0,0 +1,165 @@
+            } else if (parameters.getFromNumber().isPresent() || parameters.getToNumber().isPresent()) {
+                parameters.getFromNumber().ifPresent(x -> parts.add(parameters.getField().getSqlFieldName() + " >= ?"));
+                parameters.getToNumber().ifPresent(x -> parts.add(parameters.getField().getSqlFieldName() + " <= ?"));
+            } else {
+                parameters.getFromTime().ifPresent(x -> parts.add(parameters.getField().getSqlFieldName() + " >= ?"));
+                parameters.getToTime().ifPresent(x -> parts.add(parameters.getField().getSqlFieldName() + " <= ?"));
+            }
+            query.append(String.join(" AND ", parts));
+        }
+        return query.toString();
+    }
+    
+    /**
+     * @param query The string query template to fill.
+     * @return A full prepared statement ready to be executed.
+     * @throws SQLException - If there was an error while querying the database
+     */
+    private PreparedStatement fillTemplate(String query) throws SQLException {
+        PreparedStatement st = connection.prepareStatement(query.toString());
+        int index = 0;
+        
+        if (parameters.getSearchString().isPresent()) {
+            st.setString(++index, "%" + parameters.getSearchString().get() + "%");
+        }
+        if (parameters.getFromNumber().isPresent()) {
+            st.setInt(++index, parameters.getFromNumber().get());
+        }
+        if (parameters.getToNumber().isPresent()) {
+            st.setInt(++index, parameters.getToNumber().get());
+        }
+        if (parameters.getFromTime().isPresent()) {
+            st.setTimestamp(++index, new Timestamp(parameters.getFromTime().get().getTimeInMillis()));
+        }
+        if (parameters.getToTime().isPresent()) {
+            st.setTimestamp(++index, new Timestamp(parameters.getToTime().get().getTimeInMillis()));
+        }
+        
+        st.setInt(++index, (pageNumber - 1) * pageSize);
+        st.setInt(++index, pageSize);
+        return st;
+    }
+    
+    /**
+     * @return A string containing the ORDER BY and LIMIT sections of the query.
+     */
+    public String createSortLimitTemplate() {
+        if (sorts.size() == 0) {
+            throw new IllegalStateException("Cannot not sort data.");
+        }
+        StringBuilder statement = new StringBuilder();
+        statement.append(" ORDER BY ");
+
+        statement.append(sorts.stream()
+                .map(js -> js.sortField.getSqlFieldName() + " " + js.direction.getSql())
+                .collect(Collectors.joining(", ")));
+        
+        statement.append(" LIMIT ?, ?");
+        return statement.toString();
+    }
+    
+    /**
+     * Adds a field to be used to order the results in ascending order.
+     * 
+     * @param sortField The field to sort with.
+     */
+    public void addAscendingSort(JournalField sortField) {
+        sorts.add(new JournalSort(sortField, JournalSortDirection.ASCENDING));
```

Comment:
This method appears to be unused from anywhere except test code. It can be removed.

Reply:
Reworked how sorts are added so there is only one `addSort` method now, which is in `JournalSearchParameters`.